pandas/indexes/multi.py

```diff
@@ -656,9 +665,94 @@ def _has_complex_internals(self):
         return True
 
     @cache_readonly
+    def is_monotonic(self):
+        """
+        return if the index is monotonic increasing (only equal or
+        increasing) values.
+        """
+        return self.is_monotonic_increasing
+
+    @cache_readonly
+    def is_monotonic_increasing(self):
+        """
+        return if the index is monotonic increasing (only equal or
+        increasing) values.
+        """
+
+        # reversed() because lexsort() wants the most significant key last.
+        values = [self._get_level_values(i)
+                  for i in reversed(range(len(self.levels)))]
+        try:
+            sort_order = np.lexsort(values)
+            return Index(sort_order).is_monotonic
+        except TypeError:
+
+            # we have mixed types and np.lexsort is not happy
+            return Index(self.values).is_monotonic
```

Comment:
I presume the memory use / perf here is no worse than it was before -- where might this get called unexpectedly (causing possibly balooning memory use)?